### PR TITLE
testutil: provide a type safe helper for Mocking

### PR DESCRIPTION
**Pull request by bboozzoo**
Add an explicit `Mock()` helper for mocking single objects. Extend unit tests for exiting `Backup*` code

### CODE REVIEW

File: testutil/base.go [hunk 1]
```diff
@@ -62,8 +62,19 @@ func (s *BaseTest) AddCleanup(f func()) {
 	s.cleanupHandlers = append(s.cleanupHandlers, f)
 }
 
+// Backup a single element before further mocking.
+func Backup[T any](mockable *T) (restore func()) {
```
Comment (zyga) [inline testutil/base.go:66]: Wooooot. Thank you l! I had this in the back of my head every time I saw the old mock being used.

File: testutil/base.go [hunk 2]
```diff
@@ -62,8 +62,19 @@ func (s *BaseTest) AddCleanup(f func()) {
...
+
+	return func() {
+		*mockable = backup
+	}
+}
+
 // Backup the specified list of elements before further mocking.
-func Backup(mockablesByPtr ...interface{}) (restore func()) {
+func BackupMany(mockablesByPtr ...interface{}) (restore func()) {
+	// TODO find a type safe way of doing this when individual elements are
```
Comment (zyga) [inline testutil/base.go:76]: There is no way with varadic functions and generics. You must use a slice of restore functions IMO.

**Review by zyga — APPROVED** [review]

**Review by ZeyadYasser — APPROVED** [review]
This is great, Thanks!

**Review by MiguelPires — APPROVED** [review]
thank you

**Comment by MiguelPires** [comment]
Can we run this with SkipSpread giving it only affects tests? I won't restart it in case it was intentional

**Comment by bboozzoo** [comment]
> Can we run this with SkipSpread giving it only affects tests? I won't restart it in case it was intentional

Yes, I'll add the label and restart. I've intentionally not added the label before to have the package built with whatever Go toolchain versions are available in supported distros.

**Pull request merged**